XenAdmin/Wizards/NewVMWizard/Page_Storage.cs

```diff
@@ -412,15 +434,22 @@ private void EditButton_Click(object sender, EventArgs e)
 
                 selectedItem.Disk = dialog.NewDisk();
                 selectedItem.UpdateDetails();
-                UpdateStatusForEachDisk();
-                UpdateEnablement();
             }
+
+            UpdateStatusForEachDisk();
+            UpdateEnablement();
         }
 
 
         private void DisksGridView_SelectionChanged(object sender, EventArgs e)
         {
-            UpdateEnablement();
+            EditButton.Enabled = DisksGridView.SelectedRows.Count > 0;
+            DeleteButton.Enabled = DisksGridView.SelectedRows.Count > 0 && ((DiskGridRowItem)DisksGridView.SelectedRows[0]).CanDelete;

```

Comment:
Can we have `SelectedRows.Count` of 2 or higher? If so please check the `CanDelete` falg for all of them.

Reply:
The DataGridView has MultiSelect=false, however, this is not a bad point, we can make the code safer by checking Count==1 rather than >0.

Reply:
Updated.